ingestion/src/metadata/ingestion/source/dashboard/tableau/metadata.py

```diff
@@ -375,15 +377,19 @@ def yield_dashboard_chart(
                 if filter_by_chart(self.source_config.chartFilterPattern, chart.name):
                     self.status.filter(chart.name, "Chart Pattern not allowed")
                     continue
-                workbook_name = dashboard_details.name.replace(" ", "")
                 site_url = (
                     f"site/{self.service_connection.siteUrl}/"
                     if self.service_connection.siteUrl
                     else ""
                 )
-                chart_url = (
-                    f"#/{site_url}" f"views/{workbook_name}/" f"{chart.view_url_name}"
-                )
+                chart_url = None
+                workbook_chart_name = chart.content_url.split("/")
+                if "/sheets/" in chart.content_url and len(workbook_chart_name) > 1:
+                    chart_url = (
+                        f"#{site_url}"
+                        f"views/{workbook_chart_name[0]}/"
+                        f"{workbook_chart_name[2]}"
```

Comment:
these magic numbers look scary to me. Not sure if we can put some tests around this, or prepare a pydantic model that explains a bit better what is happening. Thank you

Reply:
yes make sense. Let me some test as well as pydantic models